src/test/kotlin/uk/gov/dwp/dataworks/logging/LogLayoutAppenderTest.kt

```diff
@@ -0,0 +1,122 @@
+        println(result)
+
+        assertJsonContainsCommonFields(result)
+        assertJsonContainsField(result, "timestamp", "1970-04-25T07:29:03.210")
+        assertJsonContainsField(result, "log_level", "WARN")
+        assertJsonContainsField(result, "message", "some message about stuff")
+        assertJsonContainsField(result, "thread", "my.thread.is.betty")
+        assertJsonContainsField(result, "logger", "logger.name.is.mavis")
+    }
+
+    @Test
+    fun `doLayout will flatten multiline messages`() {
+        val mockEvent = mock<ILoggingEvent>()
+        whenever(mockEvent.formattedMessage).thenReturn("\"some\nmessage\nabout\nstuff with\ttabs\"")
+        whenever(mockEvent.hasCallerData()).thenReturn(false)
+
+        val result = LoggerLayoutAppender.doLayout(mockEvent)
+        println(result)
+
+        assertJsonContainsCommonFields(result)
+        assertJsonContainsField(result, "message", "some | message | about | stuff with tabs")
+    }
+
+    @Test
+    fun `doLayout will format as json with embedded tuples`() {
+        val mockEvent = mock<ILoggingEvent>()
+        val embeddedTokens = semiFormattedTuples("some message about stuff", "key1" to "value1", "key2" to "value2")
+        whenever(mockEvent.formattedMessage).thenReturn(embeddedTokens)
+        whenever(mockEvent.hasCallerData()).thenReturn(false)
+
+        val result = LoggerLayoutAppender.doLayout(mockEvent)
+        println(result)
+        assertJsonContainsCommonFields(result)
+        assertJsonContainsField(result, "key1", "value1")
+        assertJsonContainsField(result, "key2", "value2")
+        assertJsonContainsField(result, "message", "some message about stuff")
+    }
+
+    @Test
+    fun `doLayout should not escape json as that would mess with our custom static log methods which do`() {
+        val mockEvent = mock<ILoggingEvent>()
+        whenever(mockEvent.formattedMessage).thenReturn("\"message-/:'!@\"")
+        whenever(mockEvent.hasCallerData()).thenReturn(false)
+
+        val result = LoggerLayoutAppender.doLayout(mockEvent)
+        assertJsonContainsField(result, "message", "message-/:'!@")
+    }
+
+    @Test
+    fun `doLayout should add exceptions when provided`() {
+        val mockEvent = mock<ILoggingEvent>()
+        whenever(mockEvent.timeStamp).thenReturn(9876543210)
+        whenever(mockEvent.level).thenReturn(Level.WARN)
+        whenever(mockEvent.threadName).thenReturn("my.thread.is.betty")
+        whenever(mockEvent.loggerName).thenReturn("logger.name.is.mavis")
+        whenever(mockEvent.formattedMessage).thenReturn("some message about stuff")
+
+        val stubThrowable = ThrowableProxy(catchMe1())
+        ThrowableProxyUtil.build(stubThrowable, catchMe2(), ThrowableProxy(catchMe3()))
+        whenever(mockEvent.throwableProxy).thenReturn(stubThrowable as IThrowableProxy)
+
+        val result = LoggerLayoutAppender.doLayout(mockEvent)
+        assertThat(result).containsPattern("\"exception\":\".*\"")

```

Comment:
Can we assert the exception text please, so we can see it is meaningful? Otherwise it might as well be "Error"??

Reply:
See [comment](https://github.com/dwp/dataworks-common-logging/pull/1#discussion_r390194553)